src/controllers/create.transaction.controller.ts

```diff
@@ -4,16 +4,22 @@ import { urlUtils } from "../utils/url";
 import { TRADING_STATUS_PATH, urlParams } from "../types/page.urls";
 import { DESCRIPTION, REFERENCE } from "../utils/constants";
 import { Session } from "@companieshouse/node-session-handler";
+import { createConfirmationStatement } from "../services/confirmation.statement.service";
+import { Transaction } from "@companieshouse/api-sdk-node/dist/services/transaction/types";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
 
   try {
     const session = req.session as Session;
     const companyNumber = req.params[urlParams.PARAM_COMPANY_NUMBER];
-    await postTransaction(session, companyNumber, DESCRIPTION, REFERENCE);
-
-    const nextPageUrl = urlUtils.getUrlWithCompanyNumber(TRADING_STATUS_PATH, companyNumber);
-    return res.redirect(nextPageUrl);
+    const transaction: Transaction = await postTransaction(session, companyNumber, DESCRIPTION, REFERENCE);
+    const transactionId = transaction.id as string;
+    const submissionResponse = await createConfirmationStatement(session, transactionId);
+    if (submissionResponse.httpStatusCode === 201) {
+      const nextPageUrl = urlUtils.getUrlWithCompanyNumber(TRADING_STATUS_PATH, companyNumber);
+      return res.redirect(nextPageUrl);
+    }
+    return next(new Error("Company is ineligible " + JSON.stringify(submissionResponse)));
```

Comment:
for the return next, it won't always be the case that it fails due to company being ineligible. Could be a null pointer in the api for example. Also the JSON.stringify can fail if there are circular dependencies in the object. I would just pick out the bits of the object you want to log, status code and the message contents.
try
`    return next(new Error(`Unable to create Confirmation Statement httpStatusCode = ${submissionResponse.httpStatusCode}, resource = ${JSON.stringify(submissionResponse.resource)}`));
`
The resource is an object we've designed so less chance of it having a circular dependency. Would be good to see if it works by changing the api to always throw an error from the create confirmation statement endpoint.

Reply:
there are backticks in the code example but github isn't showing them. replace the quotes with backticks
`    return next(new Error("Unable to create Confirmation Statement httpStatusCode = ${submissionResponse.httpStatusCode}, resource = ${JSON.stringify(submissionResponse.resource)}"));
`